[UDzPn3dS] Update readme about loading and unrestricting (#281)
neo4j/apoc: readme.adoc

[WARNING]
====
// tag::warnings[]

> APOC contains over 450 functions and procedures.

Reviewer: ```suggestion
The APOC library contains over 450 functions and procedures.
```

It is *not* recommended to load all of these into the dbms, but instead use the principle of least privilege.

> This principle dictates, that you should only load the functions and procedures that you strictly need to execute your queries.

Reviewer: ```suggestion
This principle dictates that you only load the functions and procedures necessary to execute your queries.
```

The procedures and functions that should be loaded can be specified using the Neo4j configuration setting `+dbms.security.procedures.allowlist+` in `$NEO4J_HOME/conf/neo4j.conf`.

> As an example, to load `apoc.math.maxInt` and all functions in the `apoc.rel` package, use `+dbms.security.procedures.allowlist=apoc.math.maxInt,apoc.rel.*+`.

Reviewer: ```suggestion
For example, to load `apoc.math.maxInt` and all functions in the `apoc.rel` package, use `+dbms.security.procedures.allowlist=apoc.math.maxInt,apoc.rel.*+`.
```

For security reasons, procedures that use internal APIs are disabled by default.

> Also in this case, it is recommended to use the principle of least privilege and only unrestrict those procedures which you are certain to make use of.

Reviewer: ```suggestion
In this case, it is also recommended to use the principle of least privilege and only unrestrict those procedures which you are certain to use.
```

> The procedures can be unrestricted using the Neo4j configuration setting `+dbms.security.procedures.unrestricted+` in `$NEO4J_HOME/conf/neo4j.conf` e.g. `+dbms.security.procedures.unrestricted=apoc.cypher.runFirstColumn,apoc.cypher.doIt+`.

Reviewer: ```suggestion
Procedures can be unrestricted using the Neo4j configuration setting `+dbms.security.procedures.unrestricted+` in `$NEO4J_HOME/conf/neo4j.conf`. 
For example, to unrestrict `apoc.cypher.runFirstColumn` and `apoc.cypher.doIt`, use: `+dbms.security.procedures.unrestricted=apoc.cypher.runFirstColumn,apoc.cypher.doIt+`.
```

> If you want to unrestrict a whole package of APOC functions and procedures when using the Neo4j Docker container, you need to amend e.g. `+-e NEO4J_dbms_security_procedures_unrestricted=apoc.cypher.\\\*+` to your `docker run ...` command.

Reviewer: ```suggestion
To unrestrict a whole package of APOC functions and procedures when using the Neo4j Docker container, you need to amend the Docker environment variables. 
For example, to add the full `apoc.cypher` package, add `+-e NEO4J_dbms_security_procedures_unrestricted=apoc.cypher.\\\*+` to your `docker run ...` command.
```

[WARNING]
====
// tag::warnings[]
For security reasons, procedures and functions that use internal APIs are disabled by default.

> Loading and enablement of APOC procedures and functions can be configured using the Neo4j config file.

Reviewer: ```suggestion
Loading and enabling APOC procedures and functions can be configured using the Neo4j config file.
```

> For more details , see https://neo4j.com/docs/apoc/current/installation/#restricted[the APOC installation documentation].

Reviewer: ```suggestion
For more details, see https://neo4j.com/docs/apoc/current/installation/#restricted[the APOC installation documentation].
```
